modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/topology/PartitionsEvictManager.java

```diff
@@ -133,6 +151,81 @@ public void evictPartitionAsync(CacheGroupContext grp, GridDhtLocalPartition par
         }
     }
 
+    /**
+     *
+     * @param grp Cache group context.
+     * @param parts Partitions.
+     * @throws IgniteCheckedException If failed.
+     * @return Future.
+     */
+    public IgniteInternalFuture<Void> purgePartitionsExclusively(CacheGroupContext grp,
+        List<GridDhtLocalPartition> parts) throws IgniteCheckedException {
+        validateCacheGroupForExclusivePurge(grp);
+
+        if (F.isEmpty(parts))
+            return new GridFinishedFuture<>();
+
+        GroupEvictionContext grpEvictionCtx = evictionGroupsMap.computeIfAbsent(
+            grp.groupId(), (k) -> new GroupEvictionContext(grp));
+
+        ExclusivePurgeFuture fut = new ExclusivePurgeFuture(grpEvictionCtx, parts);
+
+        synchronized (mux) {
+            if (grpEvictionCtx.exclPurgeFut != null)
+                throw new IgniteCheckedException("Only one exclusive purge can be scheduled for the same cache group." +
+                    " [grpId=" + grp.groupId() + "]");
+
+            Set<Integer> res = new HashSet<>();
+
+            for (TaskId taskId : grpEvictionCtx.taskIds)
+                res.add(taskId.part);
+
+            res.retainAll(fut.partIds);
+
+            if (!res.isEmpty())
+                throw new IgniteCheckedException("Can't schedule exclusive purge for a partition " +
```

Comment:
1. We should add {} braces for each multi-line if branch.
2. How do we expect callee handle the IgniteCheckedException we throw?